### PR TITLE
[Merged by Bors] - refactor: resolve to false if exception throws an error (PL-1360)

### PR DESCRIPTION
**Pull request by zhihil**
**Fixes or implements PL-1360**

### Brief description. What is this change?

Changing the message conditions so that if an exception occurs when evaluating the condition, we resolve the condition to `false` instead of throwing an exception. 

### Implementation details. How do you make this change?

Changed the Condition classes to accept a new callback called `logError`. We use this callback to receive error message strings from the condition and log these errors to DataDog. 

The old `log` callback passed to Condition classes was renamed to `emitTraceMessage` (no relation to DM API traces, it means "tracing" through the evaluation logic of the condition classes). The rationale here is that "logging" should specifically refer to _our_ logging to DataDog for observability, while the "logging" on the Prototype Tool is a customer-facing feature intended to help them debug their conditions. 

We need two different callbacks because error messages on DataDog vs. error messages on Prototype Tool have different audiences, us vs. the users. Therefore, the content of the error messages we log to DataDog vs. the Prototype Tool need to be different. 

Changed the Condition classes' `catch`-all clause to `return false` rather than throw an `Error`. 

### Setup information

N/A

### Deployment Notes

N/A

### Related PRs

N/A

### Checklist

- ~~[ ] Breaking changes have been communicated, including:~~
    - New required environment variables
    - Renaming of interfaces (API routes, request/response interface, etc)
- ~~[ ] New environment variables have [been deployed](https://www.notion.so/voiceflow/Add-Environment-Variables-be1b0136479f45f1adece7995a7adbfb)~~
- ~~[ ] Appropriate tests have been written~~
    - Bug fixes are accompanied by an updated or new test
    - New features are accompanied by a new test

### CODE REVIEW

File: lib/services/runtime/handlers/message/lib/conditions/expression.condition.ts [hunk 1]
```diff
@@ -114,15 +114,22 @@ export class ExpressionCondition extends BaseCondition<CompiledExpressionConditi
...
-        `an unknown error occurred executing an expression condition, details = ${JSON.stringify(
-          err,
-          null,
-          2
-        ).substring(0, 300)}`
-      );
+
+      this.log(`expression condition encountered an unexpected error and automatically resolved to 'false'`);
+
+      return false;
```
Comment (zhihil) [inline lib/services/runtime/handlers/message/lib/conditions/expression.condition.ts:132]: Main Change (1/2)

We no longer throw a formatted `Error` here and instead return `false`. The error messages will still be logged to DataDog.

File: lib/services/runtime/handlers/message/lib/conditions/script.condition.ts [hunk 2]
```diff
@@ -22,14 +22,19 @@ export class ScriptCondition extends BaseCondition<CompiledScriptCondition> {
...
-      throw new Error(
-        `an unknown error occurred executing a script condition, details = ${JSON.stringify(err, null, 2).substring(
-          0,
-          300
-        )}`
-      );
+
+      this.log(`script condition encountered an unexpected error and automatically resolved to 'false'`);
+
+      return false;
```
Comment (zhihil) [inline lib/services/runtime/handlers/message/lib/conditions/script.condition.ts:37]: Main Change (2/2)

We no longer throw a formatted `Error` here and instead return `false`. The error messages will still be logged to DataDog.

**Review by trs — APPROVED** [review]

**Pull request closed without merging**